### PR TITLE
feat: add ORDRSP D96A fixtures and validation coverage

### PR DESCRIPTION
**Pull request by errfld**
## Summary
- add `EANCOM_ORDRSP` D96A runtime schema fixture
- add ORDRSP EDI fixtures (`valid` minimal/full and `invalid` missing BGM)
- add mapping examples for `ordrsp_to_json` and `ordrsp_to_csv`
- add `edi-validation` integration tests that parse fixtures and validate against the new schema
- add `edi-adapter-edifact` as an `edi-validation` dev-dependency for fixture-driven parser tests
- record implementation learning in `AGENTS.md`

## Validation
- `cargo fmt --all`
- `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- `cargo test --workspace --all-targets --all-features`

## Issue
- part of #82 (ORDRSP track)

## Summary by CodeRabbit

* **New Features**
  * ORDRSP (order response) EDIFACT support: schema, minimal & full fixtures, and mappings to CSV/JSON.
  * Parser now correctly groups multiple line items; mapping engine supports component selectors for targeted field extraction.

* **Tests**
  * Added end-to-end integration tests for valid, full, minimal, and invalid ORDRSP scenarios, including strict/lenient validation behavior.

* **Documentation**
  * Added note about required dev-dependency for running ORDRSP validation tests.

### CODE REVIEW

**Comment by errfld** [comment]
Addressed review feedback in commit `bcaeb2a`.

Changes applied:
- Updated `normalize_document_for_validation` to return `Cow<'_, Document>` and avoid unconditional cloning.
- Added strictness-level integration coverage for invalid ORDRSP fixture (`Strict` vs `Lenient`) using `validate_with_schema`.
- Extended missing-BGM assertions to verify contextual error quality on the `MISSING_MANDATORY_SEGMENT` issue.
- Updated ORDRSP mapping paths to remove absolute `QTY`/`PRI` lookups inside `/LIN` foreach and aligned DTM/line field paths with review suggestions.

Validated with:
- `cargo test -p edi-validation --test ordrsp_integration_test`

**Comment by errfld** [comment]
Addressed the latest Codex review comments in commit `24ddc13`.

Changes applied:
- Updated ORDRSP JSON mapping date extraction to use the composite date component (`/DTM/e1/c2`).
- Updated ORDRSP CSV/JSON line mappings to use composite component paths (`LIN/e3/c1`, `QTY/e1/c2`, `PRI/e1/c2`).
- Updated ORDRSP CSV/JSON foreach source to `LINE_ITEM` with segment-scoped fields (`LIN/...`, `QTY/...`, `PRI/...`) to keep line values context-correct.
- Updated parser grouping to group ORDRSP line loops into `LINE_ITEM` segment groups (matching mapping assumptions), and added regression test coverage for multiple ORDRSP line loops.

Validation run:
- `cargo test -p edi-adapter-edifact`
- `cargo test -p edi-validation --test ordrsp_integration_test`
- `cargo test -p edi-mapping --test mapping_examples_test`

**Comment by errfld** [comment]
Addressed the remaining CodeRabbit comments in commit `be5c54e`.

Updates made:
- Refactored parser dispatch in `build_document` to use a helper:
  - Added `needs_line_item_grouping(message_type: Option<&str>) -> bool`
  - Renamed grouping function usage to `group_line_items(...)` for clearer intent
- Strengthened `test_ordrsp_grouping_with_multiple_lin_loops`:
  - Added assertion for parsed `docs[0].metadata.doc_type == Some("ORDRSP".to_string())`
  - Added second `LINE_ITEM` child assertions for `LIN`, `QTY`, and `PRI`

Validation run:
- `cargo test -p edi-adapter-edifact`

Note on the duplicate mapping-shape comments: this branch now relies on parser-produced composite component nodes (`c1/c2/...`) and tests are passing with current IR shape.

**Comment by errfld** [comment]
Addressed the latest comments in commit `ff63a18`.

What was updated:
- Added qualifier-aware path selectors in mapping runtime (`crates/edi-mapping/src/runtime.rs`), including support for path components like `DTM[c1='137']`, `QTY[c1='21']`, `PRI[c1='AAA']`.
- Updated ORDRSP mappings to use qualifier-specific paths:
  - `response_date`: `/DTM[c1='137']/e1/c2`
  - `confirmed_quantity`: `QTY[c1='21']/e1/c2`
  - `unit_price`: `PRI[c1='AAA']/e1/c2`
- Added runtime tests for selector resolution and foreach selector behavior.

Validation run:
- `cargo test -p edi-mapping`
- `cargo test -p edi-adapter-edifact`

This addresses the date/line-value mis-selection concerns caused by first-match path resolution in repeated segments.

**Comment by errfld** [comment]
Addressed the latest review comments in commit `7799c46`.

Changes applied:
- `crates/edi-adapter-edifact/src/parser.rs`
  - Fixed `needs_line_item_grouping` pattern binding shadowing (`Some(msg_type)` instead of shadowing `message_type`).
  - Added explicit line-item boundary handling helper and included `CNT` as a boundary so summary counts are not nested under the last `LINE_ITEM` when `UNS` is absent.
  - Added regression test `test_ordrsp_grouping_keeps_cnt_outside_last_line_item`.
- `crates/edi-mapping/src/runtime.rs`
  - Simplified selector `e*` branch to a single lookup and removed dead fallback path.
  - Added debug logging for unexpected selector key patterns before fallback qualifier lookup.
  - Added selector edge-case tests:
    - `test_selector_no_match_falls_back_to_null`
    - `test_selector_empty_or_malformed_selector_graceful`
- `testdata/mappings/ordrsp_to_json.yaml`
  - Added a comment documenting parser/schema coupling for `response_date` path `/DTM[c1='137']/e1/c2`.

Validation run:
- `cargo test -p edi-adapter-edifact`
- `cargo test -p edi-mapping`
- `cargo test -p edi-validation --test ordrsp_integration_test`

File: crates/edi-mapping/src/runtime.rs [hunk 1]
```diff
@@ -386,6 +394,118 @@ impl MappingRuntime {
...
+
+    fn parse_selector(selector: &str) -> Option<(&str, &str)> {
+        let trimmed = selector.trim();
+        if trimmed.is_empty() {
+            return None;
+        }
+
+        if let Some((key, raw_value)) = trimmed.split_once('=') {
+            let key = key.trim();
+            let value = raw_value.trim().trim_matches('\'').trim_matches('"');
```
Comment (errfld) [inline crates/edi-mapping/src/runtime.rs]: can we extract this?

**Comment by errfld** [comment]
@codex address that feedback

Please handle the remaining **non-outdated unresolved** review work on this PR.

Scope: `crates/edi-mapping/src/runtime.rs` (plus tests in same file/module).

Tasks:
1. Fix selector safety for unknown keys.
- Current behavior silently falls back to qualifier lookup for arbitrary keys.
- Update `selector_value`/matching flow so malformed keys (e.g. `foo`) do **not** silently match.
- Keep numeric qualifier-key behavior working for EDI code selectors (e.g. `2005`, `3035`).

2. Remove misleading invalid-selector sentinel value.
- `INVALID_SELECTOR_VALUE` is not semantically used; replace sentinel-string state with clearer typed state where practical.
- Keep behavior deterministic for malformed selectors.

3. Document and test parse-selector defaults.
- Add a short doc comment clarifying that `[='153']` and `['153']` normalize to `c1='153'`.
- Extend tests to cover both forms and ensure equal parsed behavior.

4. Add focused cN selector coverage and case handling.
- Add a unit test for `c2` dispatch (`e1/c2` lookup path).
- Ensure selector key case handling is consistent (`c2` and `C2` should behave the same).

5. Address maintainer note on extraction.
- There is a maintainer thread on `runtime.rs` asking to extract logic near selector parsing; factor duplicated selector-normalization logic into a helper if still relevant.

Acceptance criteria:
- All unresolved non-outdated runtime selector threads are satisfied with code + tests.
- Behavior remains backward-compatible for valid selector keys.
- No panics on malformed selectors.

Validation:
- `cargo fmt --all`
- `cargo clippy -p edi-mapping --all-targets -- -D warnings`
- `cargo test -p edi-mapping`

When done, commit to this PR branch and summarize exactly which threads were addressed.

**Comment by errfld** [comment]
@coderabbitai review

**Comment by errfld** [comment]
@coderabbitai review

**Comment by errfld** [comment]
Addressed the latest review blockers and re-ran local verification.

Validation commands run locally:
- cargo test -p edi-validation ordrsp_cnt_fixture_validates_against_schema -- --nocapture
- cargo test -p edi-validation ordrsp -- --nocapture
- cargo test -p edi-adapter-edifact ordrsp_grouping -- --nocapture
- cargo fmt --all -- --check
- cargo clippy --workspace --all-targets --all-features -- -D warnings
- cargo test --workspace --all-targets --all-features

@coderabbitai review
@codex review
